tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/decisiontree/RandomForest.scala

```diff
@@ -80,36 +86,38 @@ class RandomForestTrainer(validationPercentage: Double, numDecisionTrees: Int,
   maximumDepthPerTree: Int = Integer.MAX_VALUE)
     extends ProbabilisticClassifierTrainer {
 
+  private val dtTrainer = new DecisionTreeTrainer(validationPercentage, featuresExaminedPerNode,
+    maximumDepth = maximumDepthPerTree)
+
   /** Induces a RandomForest from a set of feature vectors.
     *
     * @param data a set of feature vectors to use for training
     * @return the induced random forest
     */
   override def apply(data: FeatureVectorSource): ProbabilisticClassifier = {
-
-    val dtTrainer = new DecisionTreeTrainer(validationPercentage, featuresExaminedPerNode,
-      maximumDepth = maximumDepthPerTree)
-    System.gc()
-    val initialMemory: Double = Runtime.getRuntime.totalMemory - Runtime.getRuntime.freeMemory()
-    val result = RandomForest(
-      data.allOutcomes,
-      Range(0, numDecisionTrees) flatMap { _ =>
-        val trainingData = useBagging match {
-          case true => data //.getBag
-          case false => data
-        }
-        dtTrainer(trainingData) match {
-          case dt: DecisionTree => Some(dt)
-          case _ => None
-        }
+    var subtreeFiles: Seq[File] = Seq() // TODO: do we need a var here?
+    Range(0, numDecisionTrees) foreach { _ =>
+      dtTrainer(data) match {
+        case dt: DecisionTree =>
+          val tempFile: File = File.createTempFile("temp.", ".dt")
+          Resource.using(new PrintWriter(tempFile)) { writer =>
+            writer.println(dt.toJson.compactPrint)
+          }
+          subtreeFiles = tempFile +: subtreeFiles
```

Comment:
It returns a ProbabilisticClassifier, according to their interface of ProbabilisticClassifierTrainer. This will always be a DecisionTree in the case of the DecisionTreeTrainer.